lib/connection/connection.js

```diff
@@ -427,24 +429,38 @@
    * @returns {String} the query status.
    */
   this.getQueryStatusThrowIfError = async function (queryId) {
-    const status = this.getQueryStatus(queryId);
+    const status = await this.getQueryStatus(queryId);
 
     let message, code, sqlState = null;
 
     if (this.isAnError(status)) {
       const response = await getQueryResponse(queryId);
       message = response['message'] || '';
       code = response['code'] || -1;
-
-      if (response['data']) {
-        message += response['data']['queries'].length > 0 ? response['data']['queries'][0]['errorMessage'] : '';
-        sqlState = response['data']['sqlState'];
+      sqlState = response['state'] || '';
+      let responseData = response['data'];
+
+      if (responseData) {
+        if (responseData['queries'].length > 0) {
+          const firstQuery = responseData['queries'][0];
+          message += firstQuery['errorMessage'];
+          code = firstQuery['errorCode'];
+          sqlState = firstQuery['state'];
+          responseData = {};
+          [
+            'id', 'requestId', 'status', 'state', 'sqlText', 'sessionId', 'clientSendTime', 'startTime',
+            'endTime', 'isDescribe', 'errorCode', 'errorMessage', 'warehouseName', 'queryTag'
+          ].forEach( (key) => {
```

Comment:
I assume that this is added to not lose any data?

Reply:
responded internally

Reply:
sure, we can close this one and continue with 746

Reply:
Thank you @sfc-gh-dszmolka